File: calculator.py


import subprocess

def calculator_program():
    try:
        print("Enter your input for the calculator program:")

        # Get user input for each line
        num_one = input("Please enter your first number: ")
        num_two = input("Please enter your second number: ")
        num_three = input("Please enter your third number: ")
        choice = input("Enter your choice (1 for Addition, 2 for Subtraction, 3 for Multiplication, 4 for Division): ")

        # Format the input as a multiline string
        input_data = f"{num_one}\n{num_two}\n{num_three}\n{choice}"

        # Run the subprocess with input and output pipes
        with subprocess.Popen(["./calculator"], stdin=subprocess.PIPE, stdout=subprocess.PIPE, stderr=subprocess.PIPE, text=True) as process:
            # Send input to the subprocess
            process.stdin.write(input_data + '\n')
            process.stdin.flush()  # Flush the input buffer to make sure it's sent

            # Wait for the subprocess to finish and get output
            output, error = process.communicate()

            # Print only the relevant output
            relevant_output = "\n".join(line for line in output.split('\n') if "The Ans is:" in line)
            print("Output from calculator program:")
            print(relevant_output)

            # Print any errors (if needed)
            if error:
                print("Error from calculator program:")
                print(error)

    except subprocess.CalledProcessError as e:
        print(f"Error: {e}")

if __name__ == "__main__":
    calculator_program()

File: calculator.cpp

//this is a calculator c++ program//

#include <iostream>
#include <cstdio>

int main() {
    FILE* file = fopen("output.txt", "w");

    if (file == nullptr) {
        std::cerr << "Error opening file for writing." << std::endl;
        return 1;
    }

    int Num_One, Num_Two, Num_Three, choice, Result;

    std::cout << "Welcome to calculator programmer" << std::endl;

    std::cout << "Please enter your first number" << std::endl;
    std::cin >> Num_One;

    if (std::cin.fail()) {
        std::cerr << "Error reading input. Exiting." << std::endl;
        fclose(file);
        return 1;
    }

    // Use fputs to write to the file
    fputs("Please enter your first number\n", file);
    fprintf(file, "%d\n", Num_One);

    std::cout << "Please enter your second number" << std::endl;
    std::cin >> Num_Two;

    if (std::cin.fail()) {
        std::cerr << "Error reading input. Exiting." << std::endl;
        fclose(file);
        return 1;
    }

    // Use fputs to write to the file
    fputs("Please enter your second number\n", file);
    fprintf(file, "%d\n", Num_Two);

    std::cout << "Please enter your third number" << std::endl;
    std::cin >> Num_Three;

    if (std::cin.fail()) {
        std::cerr << "Error reading input. Exiting." << std::endl;
        fclose(file);
        return 1;
    }

    // Use fputs to write to the file
    fputs("Please enter your third number\n", file);
    fprintf(file, "%d\n", Num_Three);

    std::cout << "1) Addition\n";
    std::cout << "2) Subtraction\n";
    std::cout << "3) Multiplication\n";
    std::cout << "4) Division\n";

    std::cin >> choice;

    if (std::cin.fail()) {
        std::cerr << "Error reading input. Exiting." << std::endl;
        fclose(file);
        return 1;
    }

    // Use fputs to write to the file
    fputs("User choice\n", file);
    fprintf(file, "%d\n", choice);

    switch (choice) {
        case 1: // Addition
            Result = Num_One + Num_Two + Num_Three;
            break;
        case 2: // Subtraction
            Result = Num_One - Num_Two - Num_Three;
            break;
        case 3: // Multiplication
            Result = Num_One * Num_Two * Num_Three;
            break;
        case 4: // Division
            if (Num_Two == 0) {
                // Use fputs to write to the file
                fputs("Please enter a number which is not equal to zero\n", file);
                fclose(file);
                return 0;
            } else {
                Result = Num_One % Num_Two;
                break;
            }
        default:
            // Use fputs to write to the file
            fputs("Invalid choice\n", file);
            fclose(file);
            return 0;
    }

    if (choice >= 1 && choice <= 4) {
        std::cout << "The Ans is: " << Result << std::endl;

        // Use fputs to write to the file
        fputs("The Ans is\n", file);
        fprintf(file, "%d\n", Result);
    }

    fclose(file);
    return 0;
}
